lib/modules/navigationMiddleware.js

```diff
@@ -2,45 +2,55 @@ import pathToRegex from 'path-to-regexp';
 import { METHODS } from './router';
 import * as actions from './actions';
 
+export const parseRoute = (url, routes) => {
+  let route;
+
+  for (route of routes) {
+    const [url, handler] = route;
+    const reg = pathToRegex(url);
+    const result = reg.exec(url);
+
+    if (result) {
+      return { handler, reg, result };
+    }
+  }
```

Comment:
ok, in that case we should make the destructuring (`const { handler, reg, result } = parseRoute(pathName, routes);`) handle the undefined case